crates/memory/src/region.rs

```diff
@@ -1,37 +1,62 @@
+/// Represents a contiguous region in memory.
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub struct Region {
     pub start: u32,
     pub len: u32,
 }
 
 impl Region {
+    pub fn new(start: u32, len: u32) -> Self {
+        Self { start, len }
```

Comment:
lets assert that len > 0 so that self_end / rhs_end are always bigger than the corresponding starts. the len should be determined by wiggle-generated GuestType impls, so its on us to make sure its always nonzero

Reply:
Excellent point! Done in 7b9782b.